### PR TITLE
TEST: disable NCCL and GDR

### PR DESCRIPTION
**Pull request by vspetrov**
## What
Disables NCCL and GDR in gtest

## Why ?
NCCL is not supported due to blocking team creation
GDR currently may result in a hang (known issue)

## How ?
internally sets env vars: UCC_TL_NCCL_TUNE=0 and UCX_IB_GPU_DIRECT_RDMA=no

### CODE REVIEW

File: test/gtest/common/test_ucc.cc [hunk 1]
```diff
@@ -247,6 +247,14 @@ UccJob::UccJob(int _n_procs, ucc_job_ctx_mode_t _ctx_mode, ucc_job_env_t vars) :
 {
     ucc_job_env_t env_bkp;
     char *var;
+
+    /* NCCL TL is disabled since it currently can not support non-blocking
+       team creation. */
+    vars.push_back({"UCC_TL_NCCL_TUNE", "0"});
```
Comment (bureddy) [inline test/gtest/common/test_ucc.cc:253]: is it a temporary fix to disable NCCL also? is there way to going forward fix to testing if non-blocking team create not support

Comment (vspetrov) [reply test/gtest/common/test_ucc.cc:253]: it is temporary until we figure out how to run NCCL from single process (not sure if that is possible at all). Gtest is not that well suited for that case

**Review by bureddy — APPROVED** [review]

File: test/gtest/common/test_ucc.cc [hunk 2]
```diff
@@ -247,6 +247,14 @@ UccJob::UccJob(int _n_procs, ucc_job_ctx_mode_t _ctx_mode, ucc_job_env_t vars) :
...
+    /* GDR is temporarily disabled due to known issue that may result
+       in a hang in the destruction flow */
+    vars.push_back({"UCX_IB_GPU_DIRECT_RDMA", "no"});
```
Comment (bureddy) [inline test/gtest/common/test_ucc.cc:256]: instead of disabling GDR here, is it ok to disable stop nv_peer_mem on test server?

**Review by Sergei-Lebedev — APPROVED** [review]

**Comment by vspetrov** [comment]
bot:retest

**Comment by artemry-nv** [comment]
bot:retest

**Pull request merged**